From fb2393132ddf7af47b6b5dd67b57048f4b914faa Mon Sep 17 00:00:00 2001
From: ratheesh kannoth <rkannoth@codeaurora.org>
Date: Mon, 13 Nov 2017 15:10:10 +0530
Subject: [PATCH 298/500] net :gre : Set skb->skb_iif number

skb->skb_iif should be set to GRE netdevice to accelerate
packet

Change-Id: I0af542d2fa64b87c51b9bf3803bb874299299026
Signed-off-by: ratheesh kannoth <rkannoth@codeaurora.org>
Signed-off-by: Subhash Kumar Katnpally <skatnapa@codeaurora.org>
Signed-off-by: Pavithra R <pavir@codeaurora.org>
---
 net/ipv4/ip_gre.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/net/ipv4/ip_gre.c
+++ b/net/ipv4/ip_gre.c
@@ -687,6 +687,8 @@ static netdev_tx_t ipgre_xmit(struct sk_
 	if (gre_handle_offloads(skb, !!(tunnel->parms.o_flags & TUNNEL_CSUM)))
 		goto free_skb;
 
+	skb->skb_iif = dev->ifindex;
+
 	__gre_xmit(skb, dev, tnl_params, skb->protocol);
 	return NETDEV_TX_OK;
 
@@ -770,6 +772,8 @@ static netdev_tx_t gre_tap_xmit(struct s
 	if (skb_cow_head(skb, dev->needed_headroom))
 		goto free_skb;
 
+	skb->skb_iif = dev->ifindex;
+
 	__gre_xmit(skb, dev, &tunnel->parms.iph, htons(ETH_P_TEB));
 	return NETDEV_TX_OK;
 
